default/python/AI/AIstate.py

```diff
@@ -43,10 +43,62 @@
 outpostSystemIDs = []
 
 
-# AIstate class
+class ConversionError(Exception):
+    """Exception to be raised if the conversion of a savegame state fails.
+
+    Automatically logs and chats to the host if raised.
+    """
+    def __init__(self, msg=""):
+        print_error(msg)
+
+
+def convert_to_version(state, version):
+    """Convert a savegame AIstate to the next version.
+
+    :param state: savegame state, modified in function
+    :type state: dict
+    :param version: Version to convert to
+    :type version: int
+    """
+    print "Trying to convert savegame state to version %d..." % version
+    current_version = state.get("version", -1)
+    print "  Current version: %d" % current_version
+    if current_version == version:
+        raise ConversionError("Can't convert AI Savegame to the same version.")
```

Comment:
in these three ConversionErrors I think it would be better to refer more specifically to "compatibility version" rather than just "version"